### PR TITLE
FRMW-895 Share LocalStorage for iFrame Apps Auth

### PR DESCRIPTION
**Pull request by halkon**
JIRA: https://jira.rax.io/browse/FRMW-895

### LGTMs
- [x] Dev LGTM
- [x] Dev+Vidyo LGTM
- [x] QE LGTM

By modifying the `rxLocalStorage` and `SessionStorage` services (which need a good rename) we are able to allow apps inside of an iframe to communicate with Origin.

### CODE REVIEW

**Comment by Droogans** [comment]
@halkon copy pasting failure here to save you 20 seconds or so.

```
PhantomJS 1.9.8 (Linux) encore.ui.rxApp rxApp default "before each" hook FAILED
	TypeError: 'undefined' is not an object (evaluating 'b.clear')
	    at /home/travis/build/rackerlabs/encore-ui/build/dist/encore-ui-tpls-1.49.0.min.js:10
	    at /home/travis/build/rackerlabs/encore-ui/src/components/rxApp/scripts/rxApp.spec.js:68
	    at invoke (/home/travis/build/rackerlabs/encore-ui/demo/bower_components/angular/angular.js:3965)
	    at workFn (/home/travis/build/rackerlabs/encore-ui/demo/bower_components/angular-mocks/angular-mocks.js:2177)
	    at /home/travis/build/rackerlabs/encore-ui/demo/bower_components/angular-mocks/angular-mocks.js:2163
	    at /home/travis/build/rackerlabs/encore-ui/src/components/rxApp/scripts/rxApp.spec.js:69
	undefined
```

**Comment by halkon** [comment]
@Droogans I saw it yesterday, I 😢 then figured out that it was the tests running minified, so today I'll fix it

**Comment by parlarjb** [comment]
LGTM

**Comment by Droogans** [comment]
This should be forked and maintained directly in the origin project, not shared by all consumers of the framework.

QE DNLGTM from a philosophical standpoint but marking as QE LGTM from a technical implementation standpoint.

**Comment by 100stacks** [comment]
LGTM

File: src/domainOverride.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+// Let's get the hostname only (no port information)
+var host = window.location.hostname;
+// Find out if we this hostname is prefixed
+var index = host.indexOf(prefix);
+// Get the domain without the prefix if it includes it
+var domain = host.substr((index < 0) ? 0 : index + prefix.length);
+// Override the documnet.domain that allows for explicit iframe communication
+if (domain !== 'localhost') {
+    document.domain = domain;
+}
```
Comment (halkon) [inline src/domainOverride.js]: magic

So I guess I was being too literal and I'll expand on it, but until I update the comment:

```javascript
by setting document.domain to the same domain on the renderer of an iframe, and the iframe content, both parent and child can communicate and access each other's javascript environments, allowing us with the ability to not just create an API for encore applications to communicate with `origin` but perhaps even figure out a way to share stuff between the two
```

Comment (halkon) [inline src/domainOverride.js]: @CITguy done

Comment (CITguy) [inline src/domainOverride.js]: Would that theoretically allow us to access directives defined in the child? or nah?

Comment (halkon) [inline src/domainOverride.js]: That would be some magic.   But it is not out of the question just haven't researched that far in

**Pull request merged**